armsrc/lfsampling.c

```diff
@@ -424,6 +425,107 @@ uint32_t SampleLF(bool verbose, uint32_t sample_size, bool ledcontrol) {
     BigBuf_Clear_ext(false);
     return ReadLF(true, verbose, sample_size, ledcontrol);
 }
+
+/**
+* SampleLF()/SniffLF() + ReadLF() + DoAcquisition_config() 
+* @return sampling result
+**/
+int ReadLF_realtime(bool reader_field) {
+    BigBuf_Clear_ext(false);
+    LFSetupFPGAForADC(config.divisor, reader_field);
+
+    // parameters from config and constants
+    const uint8_t bits_per_sample = config.bits_per_sample;
+    const int16_t trigger_threshold = config.trigger_threshold;
+    int32_t samples_to_skip = config.samples_to_skip;
+    const uint8_t decimation = config.decimation;
+
+    uint32_t sample_size = 64;
+    const int8_t size_threshold_table[9] = {0, 64, 64, 48, 64, 40, 48, 56, 64};
+    const int8_t size_threshold = size_threshold_table[decimation];
```

Comment:
It should be `const int8_t size_threshold = size_threshold_table[bits_per_sample];` there.

Reply:
As for the validation, the logSample() inside this function validates `the bits_per_sample`, which is not appropriate because it's time-critical. I'm considering moving it before the sampling starts.

Reply:
It doesn't matter if it's validated later.  If the value exceed 8, this code will have undefined behavior due to out-of-bounds array access.  Therefore, the validation would need to occur prior to that use as an array index.

Reply:
The `bits_per_sample` is from `config.bits_per_sample` which has been validated in `setSamplingConfig()`, so I think there is no need to check it again.